### PR TITLE
OHRM5X-1813: Develop buzz - create post API

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzShareAPITestCases.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      post:
+        id: 7
+      employee:
+        empNumber: 4
+        lastName: 'Abrahams'
+        firstName: 'Landon'
+        middleName: ''
+        employeeId: null
+        terminationId: null
+    meta: [ ]
```
Comment (ManulMax) [inline src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzShareAPITestCases.yaml:86]: TODO -  add photo post test

File: src/plugins/orangehrmBuzzPlugin/entity/BuzzPost.php [hunk 2]
```diff
@@ -22,13 +22,19 @@
...
+use OrangeHRM\Entity\Decorator\BuzzPostDecorator;
+use OrangeHRM\Entity\Decorator\DecoratorTrait;
 
 /**
+ * @method BuzzPostDecorator getDecorator()
+ *
  * @ORM\Table(name="ohrm_buzz_post")
  * @ORM\Entity
  */
 class BuzzPost
```
Comment (ManulMax) [inline src/plugins/orangehrmBuzzPlugin/entity/BuzzPost.php:34]: Refactor to the BuzzText

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzShareAPI.php [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws TransactionException
+     */
```
Comment (ManulMax) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzShareAPI.php:80]: TODO - Add API Doc for BuzzShareAPI

File: src/plugins/orangehrmBuzzPlugin/Api/EmployeeAnniversaryAPI.php [hunk 4]
```diff
@@ -39,7 +39,31 @@ class EmployeeAnniversaryAPI extends Endpoint implements CollectionEndpoint
 
     public const DATE_DIFFERENCE_MIN = 0;
     public const DATE_DIFFERENCE_MAX = 30;
+
     /**
+     * @OA\Get(
+     *     path="/api/v2/buzz/anniversaries",
+     *     tags={"Buzz/EmployeeAnniversary"},     *
+     *     @OA\Parameter(ref="#/components/parameters/sortOrder"),
+     *     @OA\Parameter(ref="#/components/parameters/limit"),
```
Comment (ManulMax) [inline src/plugins/orangehrmBuzzPlugin/Api/EmployeeAnniversaryAPI.php:48]: Change default limit

File: src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php [hunk 5]
```diff
@@ -99,4 +101,44 @@ public function getBuzzPhotoIdsByPostId(int $postId): array
...
+    {
+        $this->persist($buzzVideoPost);
+        return $buzzVideoPost;
+    }
+
+    /**
+     * @param BuzzPhoto $buzzPhoto
+     * @return BuzzPhoto
+     */
+    public function saveBuzzPhotos(BuzzPhoto $buzzPhoto): BuzzPhoto
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php:139]:
```suggestion
    public function saveBuzzPhoto(BuzzPhoto $buzzPhoto): BuzzPhoto
```

File: src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php [hunk 6]
```diff
@@ -99,4 +101,44 @@ public function getBuzzPhotoIdsByPostId(int $postId): array
...
+    {
+        $this->persist($buzzShare);
+        return $buzzShare;
+    }
+
+    /**
+     * @param BuzzLink $buzzVideoPost
+     * @return BuzzLink
+     */
+    public function saveBuzzVideo(BuzzLink $buzzVideoPost): BuzzLink
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php:129]:
```suggestion
    public function saveBuzzVideo(BuzzLink $buzzVideo): BuzzLink
```

File: src/plugins/orangehrmBuzzPlugin/Traits/Service/BuzzPostServiceTrait.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Traits\Service;
+
+use OrangeHRM\Buzz\Service\BuzzService;
+use OrangeHRM\Core\Traits\ServiceContainerTrait;
+use OrangeHRM\Framework\Services;
+
+trait BuzzPostServiceTrait
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Traits/Service/BuzzPostServiceTrait.php:26]: delete this trait

File: src/plugins/orangehrmBuzzPlugin/config/routes.yaml [hunk 8]
```diff
@@ -5,13 +5,27 @@ apiv2_buzz_feed:
...
 #       Mock API
 ##########################
-apiv2_buzz_newsfeed_posts:
-  path: /api/v2/buzz/posts
-  controller: OrangeHRM\Buzz\Controller\BuzzMockAPIController::getPosts
-  methods: [ GET ]
+#apiv2_buzz_newsfeed_posts:
+#  path: /api/v2/buzz/posts
+#  controller: OrangeHRM\Buzz\Controller\BuzzMockAPIController::getPosts
+#  methods: [ GET ]
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/config/routes.yaml:28]: revert this for the moment

File: src/plugins/orangehrmBuzzPlugin/entity/BuzzPhoto.php [hunk 9]
```diff
@@ -27,6 +27,21 @@
...
+        'image/jpg',
+        'image/pjpeg',
+        'image/png',
+        'image/x-png',
+    ];
+    public const ALLOWED_IMAGE_EXTENSIONS = [
+        'gif',
+        'jpeg',
+        'jpg',
+        'png',
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/entity/BuzzPhoto.php:42]:
```suggestion
        'png',
        'jfif',
```

File: src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzPostDecorator.php [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    public function __construct(BuzzPost $buzzPost)
+    {
+        $this->buzzPost = $buzzPost;
+    }
+
+    /**
+     * @return BuzzPost
+     */
+    public function getBuzzPost(): BuzzPost
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzPostDecorator.php:46]:
```suggestion
    protected function getBuzzPost(): BuzzPost
```

File: src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzPostDecorator.php [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        return $this->buzzPost;
+    }
+
+    /**
+     * @param int $empNumber
+     */
+    public function setEmployeeByEmpNumber(int $empNumber): void
+    {
+        $employee = $this->getReference(Employee::class, $empNumber);
+        $this->buzzPost->setEmployee($employee);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzPostDecorator.php:57]:
```suggestion
        $this->getBuzzPost()->setEmployee($employee);
```

File: src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzShareDecorator.php [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public function __construct(BuzzShare $buzzShare)
+    {
+        $this->buzzShare = $buzzShare;
+    }
+
+    /**
+     * @return BuzzShare
+     */
+    public function getBuzzShare(): BuzzShare
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/entity/Decorator/BuzzShareDecorator.php:48]:
```suggestion
    protected function getBuzzShare(): BuzzShare
```

File: src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoTest.php [hunk 13]
```diff
@@ -148,4 +163,86 @@ public function testGetBuzzPhotoIdsByPostId(): void
...
...Hk/burJm/SnPptzMTAMb24aXd7gPzgOVFm/Q9icrMBgAAaAUtnS/5xiqLuc4WuQryYWYCwPg+xOb1HycfJH+yvGiTEV/ObAAAgInUvLP2jM0ch7ZGAPabV4lOttyjrTGkxG4fswEAACZCR7n2TtuLNakZcAUzAWBCyLnqM232I3lphKb2ilcMTWJGAADAeBX8+qmWq0EOSSul68lpAEwoaeJ4geXG2i9+wC3RSpbMCAAAGCt5+HuK5BNrLO+s3UH5fgCtWbRJU+tG9UbLizapprTM9+uHMCMAAGC0pBrkBySPWG85j1k+s3vRNGYDQOsWbb7psl+EREd0p1ayZEYAAMC+yJ38j0kRs+cs5y+Pt/fWjmE2ALScLJaqbizazENayZIZAQAAe81bQvOvjfL5dh80P13o6T+B2QCQGC+IYkcWbSu14SUzAgAAdtc8GWT7OscGvTvHbABI2NAkKUSy2JFF21qvXDudOQEAAC/yfPMFB3KUzdLv7SxmA4C1RZv0R/uyG4u2+DlZQJ7HnAAAQH4iO1rGgfxki+dH5zMfAFi0vTy2dQTRfOYEAIB8KpUGp8gxyGucyEn86GJmBIAT2tpuOEA+mL7tyKJNx1f0A5uZAQAgPwrz+l8lxyBvcyAP2SEPsy9hRgA4RZpZHyhNra93ZtEWmtsvXbDwcGYGAIDs6/Tr0+Vn/wonchA/6mVGADipsdMWmm84tNP2KBUkAQDINq3AKD/zn3Qh95DjmJcxIwCcJjttk73AfN2hRdsaqjMBAJBNHaE5p1E234GcQ6tSMiMAUrNok6qNSx1atL0gxxM8ZgYAgAwt1oKoQ37Gb3ViZy0w/cwIgJRxqk9b80y5GWyrVg9ibgAASC99MCxHD+vu5BjRfzIrAFK7aJOdtgGnFm1B/HOpInUscwMAQPpoJUj5Wf59h4qcGWYFQOppbzS3Fm3R08VK9CFmBgCAFC3WevpPkAXS7x06uXMFswIgM+Rp2Fz5cNvu0MJtK022AQBIB8+vnys/u9e5kkdIn7U+ZgVAFhdtn3XlcvBO47pZlWgqswMAgIv0TnzjpM42V3IHSvcDyDQpRPIp+bB7TMbjrgyp7LRMntwdwewAAOCO9mr1YFkcXePQQ94dcq2izMwAAAAAyLXS/L7DvCC6yaUHvHJ/rsTMICn/DwoA3TzhnSpVAAAAAElFTkSuQmCC');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoTest.php:241]: you can put a image inside fixture folder and read it

File: src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoTest.php [hunk 14]
```diff
@@ -39,6 +47,13 @@ protected function setUp(): void
 
     public function testGetBuzzFeedPosts(): void
     {
+        $dateTimeHelper = $this->getMockBuilder(DateTimeHelperService::class)
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoTest.php:50]: ??
I feel should move to setup

File: src/plugins/orangehrmPimPlugin/entity/Employee.php [hunk 15]
```diff
@@ -1566,4 +1566,84 @@ public function setSupervisors(iterable $supervisors): void
...
+        return $this->likeOnComment;
+    }
+
+    /**
+     * @param iterable $likeOnComment
+     */
+    public function setLikeOnComment(iterable $likeOnComment): void
+    {
+        $this->likeOnComment = $likeOnComment;
+    }
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/entity/Employee.php:1648]: remove

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzShareAPI.php [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+use OrangeHRM\Entity\BuzzPhoto;
+use OrangeHRM\Entity\BuzzPost;
+use OrangeHRM\Entity\BuzzShare;
+use OrangeHRM\ORM\Exception\TransactionException;
+use Exception;
+
+class BuzzShareAPI extends Endpoint implements CollectionEndpoint
+{
+    use EntityManagerHelperTrait;
+    use BuzzPostServiceTrait;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzShareAPI.php:48]:
```suggestion
    use BuzzServiceTrait;
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmBuzzPlugin/entity/BuzzPost.php (hunk 2 above)
Comment (RajithaKumara) [reply src/plugins/orangehrmBuzzPlugin/entity/BuzzPost.php:34]: Will skip this

**Pull request merged**